x-pack/plugins/infra/public/observability_logs/log_stream_query_state/src/time_filter_state_service.ts

```diff
@@ -0,0 +1,192 @@
+        return {
+          type: 'TIME_FROM_TIME_FILTER_SERVICE_CHANGED',
+          ...timeState,
+        };
+      })
+    );
+
+const getTimefilterState = (timeFilterService: TimefilterContract): TimefilterState => ({
+  timeRange: timeFilterService.getTime(),
+  refreshInterval: timeFilterService.getRefreshInterval(),
+});
+
+export const updateTimeContextFromTimeFilterService = actions.assign(
+  (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
+    if (
+      event.type === 'TIME_FROM_TIME_FILTER_SERVICE_CHANGED' ||
+      event.type === 'INITIALIZED_FROM_TIME_FILTER_SERVICE'
+    ) {
+      return {
+        ...getTimeFromEvent(context, event),
+        refreshInterval: event.refreshInterval,
+      };
+    } else {
+      return {};
+    }
+  }
+);
+
+export const updateTimeContextFromUrl = actions.assign(
+  (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
+    if (event.type === 'INITIALIZED_FROM_URL') {
+      return {
+        ...('timeRange' in event && event.timeRange ? { ...getTimeFromEvent(context, event) } : {}),
+        ...('refreshInterval' in event && event.refreshInterval
+          ? { refreshInterval: event.refreshInterval }
+          : {}),
+      };
+    } else {
+      return {};
+    }
+  }
+);
+
+export const updateTimeContextFromTimeRangeUpdate = actions.assign(
+  (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
+    if ('timeRange' in event && event.type === 'UPDATE_TIME_RANGE') {
+      return getTimeFromEvent(context, event);
+    } else {
+      return {};
+    }
+  }
+);
+
+export const updateTimeContextFromRefreshIntervalUpdate = actions.assign(
+  (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
+    if (
+      'refreshInterval' in event &&
+      'refreshInterval' in context &&
+      event.type === 'UPDATE_REFRESH_INTERVAL'
+    ) {
+      const pause = event.refreshInterval.pause ?? context.refreshInterval.pause;
+      const value = event.refreshInterval.value ?? context.refreshInterval.value;
+
+      const nowTimestamp = Date.now();
+
+      const draftContext = {
+        refreshInterval: {
+          pause,
+          value,
+        },
+        ...(!pause
+          ? {
+              timeRange: {
+                ...DEFAULT_REFRESH_TIME_RANGE,
+                lastChangedCompletely: nowTimestamp,
+              },
+            }
+          : {}),
+        ...(!pause
+          ? {
+              timestamps: {
+                startTimestamp: datemathToEpochMillis(DEFAULT_REFRESH_TIME_RANGE.from, 'down') ?? 0,
+                endTimestamp: datemathToEpochMillis(DEFAULT_REFRESH_TIME_RANGE.to, 'down') ?? 0,
+                lastChangedTimestamp: nowTimestamp,
+              },
+            }
+          : {}),
+      };
+
+      return draftContext;
+    } else {
+      return {};
+    }
+  }
+);
+
+const getTimeFromEvent = (context: LogStreamQueryContext, event: LogStreamQueryEvent) => {
```

Comment:
➡️ [Previous logic](https://github.com/elastic/kibana/blob/8.6/x-pack/plugins/infra/public/containers/logs/log_position/log_position_state.ts#L158) (note: updates can be partial).